src/main/java/us/tastybento/bskyblock/Settings2.java

```diff
@@ -0,0 +1,1115 @@
+package us.tastybento.bskyblock;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+
+import org.bukkit.entity.EntityType;
+import org.bukkit.inventory.ItemStack;
+import org.bukkit.potion.PotionEffectType;
+
+import us.tastybento.bskyblock.api.configuration.ConfigEntry;
+import us.tastybento.bskyblock.api.configuration.ConfigEntry.GameType;
+import us.tastybento.bskyblock.api.configuration.ISettings;
+import us.tastybento.bskyblock.api.flags.Flag;
+import us.tastybento.bskyblock.config.StoreAt;
+import us.tastybento.bskyblock.database.BSBDatabase.DatabaseType;
+
+/**
+ * All the plugin settings are here
+ * @author Tastybento
+ */
+@StoreAt(filename="config.yml") // Explicitly call out what name this should have.
+public class Settings2 implements ISettings<Settings2> {
```

Comment:
ISettings<T> is a good idea, but doesn't it look redundant with the name of the class?